dask/dataframe/_hyperloglog.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*
+u"""Implementation of HyperLogLog
+
+This implements the HyperLogLog algorithm for cardinality estimation, found
+in
+
+    Philippe Flajolet, Éric Fusy, Olivier Gandouet and Frédéric Meunier.
+        "HyperLogLog: the analysis of a near-optimal cardinality estimation
+        algorithm". 2007 Conference on Analysis of Algorithms. Nice, France
+        (2007)
+
+"""
+from __future__ import division
+
+import numpy as np
+import pandas as pd
+
+def compute_first_bit(a):
+    # Consider making this less memory-hungry
+    #bits = np.bitwise_and.outer(a, 1<<np.arange(31, -1, -1))
+    bits = np.bitwise_and.outer(a, 1<<np.arange(32))
+    bits = bits.cumsum(axis=1).astype(np.bool)
+    return 33 - bits.sum(axis=1)
+
+def terrible_dataframe_hash(df):
```

Comment:
> There are also multiple reasons why multiprocessing on Unix would have exec'ed. (I don't know if they can actually apply to the multiprocessing mode of dask.)

Just for completeness (you might already know all this): Dask just uses an instance of `multprocessing.Pool` to do all the work. Prior to 3.4, anything that wasn't windows just used `fork` to spin up a process. After 3.4, the option to spawn processes with an entirely new interpreter state on non-windows was added to match the behavior of windows. The default is still to use `fork` non-windows. This behavior can be set globally using `set_start_method`, or locally by creating a context. If the user does `set_start_method('spawn')` then the random seeds will differ (`forkserver` is the other option, and that also results in shared seeds).

I'm not sure how best to handle this. A few options:

- Use `hash`, throw a warning when using the multiprocessing scheduler if the conditions are possibly faulty. Something like:

```python
if (os.getenv('PYTHONHASHSEED') is None and
    (sys.platform == 'win32' or
     (hasattr(mp, 'get_start_method') and mp.get_start_method() == 'spawn'))):
    warning.warn(SomeInformativeWarning())
```
- Use `hashlib` instead, even though it's slower and doesn't work on non-strings

My preference is for the first one - I think it's reasonable to expect users to be able to set the hashseed, especially if the warning provided is detailed enough. I think the downsides of moving off of hash outweigh the benefits that something like hashlib might provide.

Of course that's just one opinion, it'd be useful to get feedback from others here. Ping @mrocklin?